starknet_devnet/starknet_wrapper.py

```diff
@@ -734,7 +734,11 @@ async def get_class_by_address(
         state = await self.__get_query_state(block_id)
         cached_state = state.state
         class_hash = await self.get_class_hash_at(contract_address)
-        return cached_state.contract_classes[class_hash]
+
+        if class_hash in cached_state.contract_classes:
+            return cached_state.contract_classes[class_hash]
+
+        return self._contract_classes[class_hash]
```

Comment:
Can this fail? I think `self.get_class_hash_at` (done two lines above), will default to polling origin and fail correctly if the contract is not deployed on devnet or forking origin. But what if it's deployed on origin (meaning `self.get_class_hash_at` succeeds), but the class does not exist locally (meaning no key in `contract_classes`)? If this is indeed an issue, it can be addressed in a separate issue/PR.

Reply:
I was thinking about that but that would mean that we have an error in another place in the code right?

Reply:
So if the contract is forked it will be in `cached_state.contract_classes` or `self._contract_classes`?

Reply:
I think that Clement from kakarot team had issue with `get_full_contract` and forking